### ISSUE TITLE
Definition of multiple devices using nested values

### ISSUE THREAD

**Issue by PKedra** [StackStorm-Exchange/stackstorm-clicrud#1]
Could you please extend config file to allow multiple separated devices/group settings using nested values.
I mean something like this:

```
---
environments:
  default:
    username: "admin"
    password: "password"
    enable:   "password"
    method:    "ssh"
  icxcore:
    username: "admin"
    password: "password"
    enable:   "password"
    method:    "ssh"
  mlxedge:
    username: "admin"
    password: "password"
    enable:   "password"
    method:    "ssh"

```
Etc

**Comment by LindsayHill** [comment]
Adding values like that to the configuration schema alone won't be enough. How would you plan to use them? The actions don't currently take an environment value.

Perhaps another way to handle this would be to update the actions to accept username/password/enable/transport as optional variables, and use those to over-ride the defaults if specified?

**Comment by PKedra** [comment]
Fully agree - this will require additional parameter for each action - "environment". But this is a way more effective than to use username/password/enable/transport and maybe some more options in the future for each action.

**Comment by LindsayHill** [comment]
What about for users that don't need/want different environments? Should all of this logic be in the base `clicrud` action, or does it belong in a higher-level workflow?

Maybe you could push some rough/work in progress code to show how you'd like those actions to work, and what action metadata might look like? (Code doesn't have to work, but should show shape of how it might be used)

**Comment by PKedra** [comment]
For such users there is not to much changes.. They can define "default" profile and default profile may be"default" value for environment parameter for each action. 
I was working with vsphere integration pack:
https://github.com/StackStorm-Exchange/stackstorm-vsphere
and config file for this package is fully inline with my suggestion - you can define multiple vsphere servers in one config file and then to use profile name for actions. 
Biggest power of CLICRUD package is that you can use it for different devices from different vendors - this mean different profiles/environments.

**Comment by LindsayHill** [comment]
> Biggest power of CLICRUD package is that you can use it for different devices from different vendors - this mean different profiles/environments.

CLICrud doesn't support multiple vendors - only Brocade

**Comment by PKedra** [comment]
Even with Brocade only - ICX, VDX, MLXe  are slightly different and if you imagine single customer with all Brocade gears in the house, they may want to have different profiles for campus (ICX) different for DC (VDX) and yet another one for routers.

**Comment by LindsayHill** [comment]
Can you push some rough code that does what you want? Doesn't have to actually work, but should show an outline of how it would work

**Comment by DavidJohnGee** [comment]
Hi all.

I've finally got around to this. I've added configuration groups (same style as the vsphere pack) to CLICRUD and I've removed some other things, which needed tidying up a bit. In addition, base64 encoded enable and password fields have also been added as a light security feature.

Looks like Circle-CI has issues with some of my styling, so will mend those and try to push again.

I've tested it locally and it works. As soon as I can push, I will.

Thanks,
D
